### PR TITLE
Added ReadOnlyBytes.CopyTo tests and fixed bugs

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
cc: @ahsonkhan, @shiftylogic, @joshfree

### CODE REVIEW

File: src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs [hunk 1]
```diff
@@ -181,14 +181,16 @@ int IndexOfRest(byte value, int firstLength)
...
             {
-                first.Slice(buffer.Length).CopyTo(buffer);
+                first.Slice(0, buffer.Length).CopyTo(buffer);
                 return buffer.Length;
             }
             first.CopyTo(buffer);
-            // TODO (pri 2): do we need to compute the length here?
-            return first.Length + _rest.CopyTo(buffer.Slice(first.Length, ComputeLength() - first.Length));
+            if (buffer.Length == firstLength || _rest == null) return firstLength;
+            if (_rest == null) return firstLength;
```
Comment (shiftylogic) [inline src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs]: This is already covered in above case.

Comment (KrzysztofCwalina) [inline src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs]: Ah, thanks. Will fix.

File: src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs [hunk 2]
```diff
@@ -181,14 +181,15 @@ int IndexOfRest(byte value, int firstLength)
...
             {
-                first.Slice(buffer.Length).CopyTo(buffer);
+                first.Slice(0, buffer.Length).CopyTo(buffer);
                 return buffer.Length;
             }
             first.CopyTo(buffer);
-            // TODO (pri 2): do we need to compute the length here?
-            return first.Length + _rest.CopyTo(buffer.Slice(first.Length, ComputeLength() - first.Length));
+            if (buffer.Length == firstLength || _rest == null) return firstLength;
+            return firstLength + _rest.CopyTo(buffer.Slice(firstLength));
```
Comment (ahsonkhan) [inline src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs]: Will firstLength always be >= buffer.Length? It won't eve be less?

Comment (KrzysztofCwalina) [inline src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs]: firstLength can be >, <, or == buffer.Length.

**Pull request merged**